### ISSUE TITLE
does darknet support multiple types of class？

### ISSUE THREAD

**Issue by lqian** [AlexeyAB/darknet#2475]
any possible to detect object with multiple attributes by darknet? for example. detect person's gender, age and clothes color in only one network.

**Comment by AlexeyAB** [comment]
Yes, Yolo v3 uses multi-label classifications, so you can mark each object with multiple thruth bboxes with different classes, and it will be detected as 1 bbox with multiple classes.

**Comment by lqian** [comment]
@AlexeyAB could you please give a network config sample of yolov3?

**Comment by lqian** [comment]
the proposal method neither modify source code nor refine network config file. just make gender label, age label and clothes color label for a person. is it right?

**Comment by AlexeyAB** [comment]
@lqian Yes. Here everything is correct.

Just ensure that you use Yolo v3 cfg file with `[yolo]` layers instead of Yolo v2 with `[region]` layers.

**Comment by lqian** [comment]
does yolov3 support regression age while detecting person?

**Comment by lqian** [comment]
@AlexeyAB 
some region always have 0 sample with this method. i.e. 
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000004, .5R: -nan, .75R: -nan,  count: 0

**Comment by AlexeyAB** [comment]
@lqian Just there are not objects in your training dataset for these anchors in the last yolo layer.

**Comment by lqian** [comment]
the model got 93.52% recall after train 23000 iteration customize dataset with a minor modification with yolov3.cfg. thanks a lot!

**Comment by TinyQi** [comment]
hi, @AlexeyAB.I have some problems that have been bothering me for a long time. Now i know yolov3 can train multi-label classifications. But when I tried to use focal loss  to train the model, the results were strange.Every class's prob became to highest level.Another version model without focal loss is ok.Am i do something wrong?

**Focal Loss**
dets[0] prob list:class[0]:0.990  class[1]:0.991  class[2]:0.988  class[3]:0.992  
dets[1] prob list:class[0]:0.997  class[1]:0.998  class[2]:0.996  class[3]:0.998  
dets[2] prob list:class[0]:0.991  class[1]:0.992  class[2]:0.988  class[3]:0.993  
dets[3] prob list:class[0]:0.997  class[1]:0.998  class[2]:0.995  class[3]:0.998  
dets[0] prob list:class[0]:0.997  class[1]:0.998  class[2]:0.997  class[3]:0.998  
dets[1] prob list:class[0]:0.997  class[1]:0.998  class[2]:0.996  class[3]:0.998  
dets[2] prob list:class[0]:0.996  class[1]:0.996  class[2]:0.995  class[3]:0.997  
dets[3] prob list:class[0]:0.997  class[1]:0.998  class[2]:0.996  class[3]:0.998  
dets[4] prob list:class[0]:0.998  class[1]:0.998  class[2]:0.996  class[3]:0.998  
dets[5] prob list:class[0]:0.997  class[1]:0.997  class[2]:0.995  class[3]:0.998  
dets[0] prob list:class[0]:0.826  class[1]:0.827  class[2]:0.826  class[3]:0.827  
dets[1] prob list:class[0]:0.995  class[1]:0.995  class[2]:0.994  class[3]:0.996


**No Focal Loss**
dets[0] prob list:class[0]:0.007  class[1]:0.453  class[2]:0.001  class[3]:0.456  
dets[1] prob list:class[0]:0.015  class[1]:0.960  class[2]:0.002  class[3]:0.970  
dets[2] prob list:class[0]:0.012  class[1]:0.987  class[2]:0.002  class[3]:0.992  
dets[3] prob list:class[0]:0.009  class[1]:0.991  class[2]:0.001  class[3]:0.999  
dets[0] prob list:class[0]:0.006  class[1]:0.860  class[2]:0.001  class[3]:0.864  
dets[1] prob list:class[0]:0.007  class[1]:0.949  class[2]:0.001  class[3]:0.954  
dets[2] prob list:class[0]:0.007  class[1]:0.900  class[2]:0.001  class[3]:0.908  
dets[3] prob list:class[0]:0.005  class[1]:0.998  class[2]:0.001  class[3]:0.999  
dets[4] prob list:class[0]:0.006  class[1]:0.997  class[2]:0.001  class[3]:0.999  
dets[5] prob list:class[0]:0.006  class[1]:0.996  class[2]:0.001  class[3]:0.998  
dets[0] prob list:class[0]:0.010  class[1]:0.665  class[2]:0.001  class[3]:0.673
